lib/internal/bootstrap/node.js

```diff
@@ -622,6 +622,30 @@
       // It's possible that defaultTriggerAsyncId was set for a constructor
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
+      // if node-report is enabled, call into its handler to see
+      // whether it is interested in handling the situation.
+      // Ignore if the error is scopped inside a domain.
+      // use == in the checks as we want to allow for null and undefined
+      if (er == null || er.domain == null) {
+        try {
+          const report = internalBinding('node_report');
+          if (report != null) {
+            if (internalBinding('options').getOptions(
+              '--experimental-report')) {
```

Comment:
A while after this PR was first opened (:smile:), we’ve started using `getOptionValue()` from `internal/options` for this – it has the advantage of caching options in JS land, and being a bit faster because of that

Reply:
above comment (unable to in-line reply to that for some reason):
> 
I’m not seeing this anywhere? I mean, it’s just a typo, but it would be nice to address it :)

sorry, can you pls elaborate? which one did you mean? the handler is called right inside the block (`report.onUnCaughtException`)

or is it related to the `domain` statement? or something else?

re: `getOptionValue` : for some reason, it does not seem to work in this file, while it works fine in `lib/util.js` . It is always returning false! I am not sure why: Is it that has something to do with the loading sequence of the module vs. this file?

Reply:
Yeah, that might be because it’s this file specifically … maybe leave a TODO comment?

Reply:
I think `getOptions` only returns `{options, alias}` by now (it doesn't take any arguments), so this check is essentially `if (obj)` which is always true? Can you do something like `console.log(NativeModule.require('internal/options').options)` here to see if the options are added properly?

Reply:
@joyeecheung - thanks, I get this among others:
```
'--experimental-report' => { helpText: 'enable node report generation',
    envVarSettings: 0,
    type: 2,
    value: true },
```

Is everything ok here?

Reply:
@gireeshpunathil In that case I think `getOptionValue('--experimental-report')` is supposed to work since it simply queries that map>

Reply:
ah! please ignore this! Now it works!!

Did something got landed in the last few days? the only change in my side is a rebase.

Reply:
and pushed a change to that effect, effectively removing the TODO.
(Going to resolve some old comments, it is becoming trickier to manage in the browser)